src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,51 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private static final int CAPACITY = 10;
+    private static final double COEFFICIENT = 0.5;
+    private static final int NUMBER_ONE = 1;
+    private T[] array;
+    private int size;
 
+    public MateStack() {
+        array = (T[]) new Object[CAPACITY];
+        size = 0;
+    }
+
+    public void push(T value) {
+        if (size >= array.length) {
+            array = resize(array);
+        }
+        array[size] = value;
+        size++;
     }
 
     public T peek() {
-        return null;
+        if (size == 0) {
+            throw new EmptyStackException();
+        }
```

Comment:
```suggestion
        if (size == 0) {
            throw new EmptyStackException();
        }
```
add this in separate method please

Reply:
done